docs/source/developing/writing-docs.adoc

```diff
@@ -12,85 +12,156 @@ toc::[]
 
 This section contains information about contributing to the Minishift documentation.
 
-[[section-building-minishift-docs]]
-== Building Minishift documentation
+[[contribute-to-docs]]
+== Contributing to the documentation
 
-Minishift documentation is located in the `docs` sub-folder. The
+Minishift is an open-source project and we welcome contributions. Similar to code
+contributions, if you want to add or edit Minishift documentation, you can create an
+issue in our link:https://github.com/minishift/minishift/issues[Github issue tracker] and
+submit a pull request with the changes or request assistance from our technical writers.
+
+[[docs-conventions-guidelines]]
+== Conventions and guidelines
+
+The Minishift documentation is authored in the AsciiDoc markup format. Since Minishift is
+closely related to the OpenShift Origin project, we follow the same conventions for
+tagging, formatting, and structure.
+
+The documentation source structure follows a modular format, where each file is called
+a _topic_ and each topic is stored under a sub-folder based on the relevant category, such as
+getting started, using Minishift, or command reference.
+
+TIP: You can check out the raw file for this topic or other existing Minishift topics to see
+how information is structured and which tags to use.
+
+[[common-conventions]]
+=== Commonly-used conventions
+
+- Each file must contain a mandatory header section with metadata that is required for publication.
+The only exception to this is the command reference topics, which are automatically generated and
+do not require this header.
+
+- Whenever we link to internal documentation topics, you should use relative links that can
+be resolved during the publication workflow instead of hard-coding URLs.
+
+- Relative links must include breadcrumbs from the *_docs/source_* sub-folder to the topic,
+to make sure that the generated URL is correct, even if you are linking from the same sub-folder.
+
+- Because code blocks must be parsed through multiple toolchains, you should avoid complex
```

Comment:
What exactly is not supported? So we have basically two main use cases atm. Console listings and YAML/JSON config. The latter should imo be marked with [source,yaml|json]. Code formatting for these is in place and also used within the existing docs.openshift.org. 

Regarding console listings, I am in favour with marking them with [listing.console]. This makes things more explicit in the source and has not effect in the rendered docs. 

> Because code blocks must be parsed through multiple toolchains, you should avoid complex

what are the multiple toolchains and which concrete asciidoctor syntax should be avoided?